include/remill/Arch/Instruction.h

```diff
@@ -254,6 +257,148 @@ class Instruction {
     kCategoryConditionalAsyncHyperCall,
   } category;
 
+
+  class Flow {};
+
+  class DirectFlow : Flow {
+   public:
+    DirectFlow() = delete;
+    DirectFlow(uint64_t known_target, DecodingContext static_context);
+
+    uint64_t known_target;
+    DecodingContext static_context;
+
+    bool operator==(const DirectFlow &rhs) const;
+  };
+
+  class IndirectFlow : Flow {
+   public:
+    // We may have info in the decoder that tells us a context value
+    std::optional<DecodingContext> maybe_context;
+
+    IndirectFlow() = delete;
+    IndirectFlow(std::optional<DecodingContext> maybe_context);
+
+    bool operator==(const IndirectFlow &rhs) const;
+  };
+
+  class FallthroughFlow : Flow {
```

Comment:
Inconsistent orderding in classes - choose one way - attrs, ctors, methods or something else but keep consistent.